entry_types/scrolled/package/src/frontend/useIframeHeight.js

```diff
@@ -0,0 +1,22 @@
+import {useState, useEffect} from 'react'
+
+export function useIframeHeight(url) {
+  const [height, setHeight] = useState('400px');
+
+  useEffect(() => {
+    window.addEventListener('message', receive)
+
+    function receive(event) {
+      if (typeof event.data['datawrapper-height'] !== 'undefined') {
+        for (var chartId in event.data['datawrapper-height']) {
+          if (url.indexOf(chartId) > -1) {
+            setHeight(event.data['datawrapper-height'][chartId] + 'px');
+          }  
+        }
+      }
+    }
+    return () => document.removeEventListener('message', receive);
+  }, [url]);
+
+  return height
+}
```

Comment:
Let's move this hook to `src/contentElements/dataWrapperChart` since it's so specific to this use case. 

Let's also add a test for the hook. It can live in `spec/contentElements/dataWrapperChart` (the `spec/contentElements` directory does not yet exist since the other content elements do not have tests. They mostly assemble functionality from `src/frontend` that is already tested there. But in this case we have some content element specific logic in this hook that can be tested individually). You can follow a similar approach like the [spec of `usePostMessageListener`](https://github.com/codevise/pageflow/blob/master/entry_types/scrolled/package/spec/frontend/usePostMessageListener-spec.js#L7). (We could have also considered reusing that hook here, but it currently only registers listeners when the page is rendered in an iframe itself to receive messages from the parent. So I'd say we keep our new hook exactly the way it is) Note that [jsdom dispatches messages asynchronously](https://github.com/jsdom/jsdom/blob/574eb51a29c17c01e6529dda4cdbdb96222d01c2/lib/jsdom/living/post-message.js#L35). This is why the test uses the `tick` helper to wait until the message has been processed.